### ISSUE TITLE
FB.ui sharing doesn't work

### ISSUE THREAD

**Issue by JustinWestphal** [earnolmartin/jquery_social_media_share_plugin#2]
Only the sharer.php link works, and even that won't take arguments such as description or image, only URL. In the if/else statement, if(window.FB) never triggers (even with the FB app ID set in the beginning of the file), and the loadSDKifNeeded function does nothing either.

**Comment by earnolmartin** [comment]
Works fine for me.  I just tried it on my implementation and on the example site.  Make sure you have a valid AppID.  The loadFacebookSDKIfNeeded function will download the SDK Javascript library from Facebook assuming you have provided a valid AppId.  This library must be loaded in order for Facebook sharing to work.  

What browser are you using?  Does the SDK script show up in the developer tools for the browser you're using? What version of jQuery are you using?

**Comment by jcw-** [comment]
Yeah, I had some issues with this as well, it was because I was already loading the FB SDK (just to show a FB post in the content, not to actually do anything fancy) and so it would end up not loading it (which is correct), but then it would think I was wanting to use the FB.ui technique, which doesn't actually work unless you have registered your website as a proper FB app and the user allows the website to connect to their profile and all that (which I wasn't doing, I still wanted to use the sharer.php technique).

So I ended up modify the library to continue using the sharer.php technique if the app id wasn't specified, by modifying line 164 to this (I'm not setting the App Id):

> if(window.FB && settings.facebookAppId != ""){

It is true that the sharer.php method only takes a URL, but that is a bit misleading because Facebook will actually check the linked content for meta "og" properties to get a description and image.

**Comment by earnolmartin** [comment]
Good catch.  I will put that in there.  Thank you for opening this issue!
